File: frontend/crop_logos.py
# pyrefly: ignore [missing-import]
import cv2
# pyrefly: ignore [missing-import]
import numpy as np

def make_transparent(img, bg_color_bgr, tolerance=25):
    # Convert BGR to BGRA
    bgra = cv2.cvtColor(img, cv2.COLOR_BGR2BGRA)
    
    # Create mask for background color
    lower_bound = np.array([max(0, c - tolerance) for c in bg_color_bgr])
    upper_bound = np.array([min(255, c + tolerance) for c in bg_color_bgr])
    
    mask = cv2.inRange(img, lower_bound, upper_bound)
    
    # Set alpha channel to 0 where background matches
    bgra[mask > 0, 3] = 0
    
    return bgra

def process():
    image_path = r"C:\Users\asus\.gemini\antigravity-ide\brain\195497f5-159f-486a-a73a-294c130a7320\media__1781627568197.jpg"
    img = cv2.imread(image_path)
    if img is None:
        print("Failed to load image")
        return
        
    h, w, _ = img.shape
    print(f"Loaded image size: {w}x{h}")
    
    # Crop top half (Light Mode Logo)
    top_half = img[0:h//2, 0:w]
    # Crop bottom half (Dark Mode Logo)
    bottom_half = img[h//2:h, 0:w]
    
    # For Light Mode, background is white (typically around [255, 255, 255])
    # Let's read top-left pixel for exact background BGR
    light_bg = top_half[5, 5]
    print(f"Light logo background BGR: {light_bg}")
    logo_light = make_transparent(top_half, light_bg, tolerance=30)
    
    # For Dark Mode, background is navy (e.g. around [51, 17, 10])
    dark_bg = bottom_half[5, 5]
    print(f"Dark logo background BGR: {dark_bg}")
    logo_dark = make_transparent(bottom_half, dark_bg, tolerance=25)
    
    # Crop tightly to the content (remove empty transparent margins for better layout scaling)
    def crop_content(png_img):
        # Find all non-transparent pixels
        alpha = png_img[:, :, 3]
        pts = np.argwhere(alpha > 0)
        if len(pts) == 0:
            return png_img
        y1, x1 = pts.min(axis=0)
        y2, x2 = pts.max(axis=0)
        # Add a small padding of 5 pixels
        h_img, w_img, _ = png_img.shape
        y1 = max(0, y1 - 5)
        x1 = max(0, x1 - 5)
        y2 = min(h_img, y2 + 5)
        x2 = min(w_img, x2 + 5)
        return png_img[y1:y2, x1:x2]

    logo_light_cropped = crop_content(logo_light)
    logo_dark_cropped = crop_content(logo_dark)
    
    # Save the files
    light_out = r"c:\Users\asus\OneDrive\Desktop\eazzio-reminder\frontend\assets\images\logo_light.png"
    dark_out = r"c:\Users\asus\OneDrive\Desktop\eazzio-reminder\frontend\assets\images\logo_dark.png"
    
    cv2.imwrite(light_out, logo_light_cropped)
    cv2.imwrite(dark_out, logo_dark_cropped)
    print("Saved light logo to:", light_out)
    print("Saved dark logo to:", dark_out)

if __name__ == "__main__":
    process()

File: frontend/crop_logos.dart
import 'dart:io';
import 'package:image/image.dart' as img;

img.Image makeTransparent(img.Image src, img.Pixel bg, {int tolerance = 30}) {
  final result = src.numChannels == 4 ? src : src.convert(numChannels: 4);
  
  final bgR = bg.r;
  final bgG = bg.g;
  final bgB = bg.b;
  
  for (final frame in result.frames) {
    for (final pixel in frame) {
      final pr = pixel.r;
      final pg = pixel.g;
      final pb = pixel.b;
      
      final diffR = (pr - bgR).abs();
      final diffG = (pg - bgG).abs();
      final diffB = (pb - bgB).abs();
      
      if (diffR <= tolerance && diffG <= tolerance && diffB <= tolerance) {
        pixel.a = 0;
      }
    }
  }
  return result;
}

img.Image cropContent(img.Image src, {int padding = 5}) {
  int minX = src.width;
  int maxX = 0;
  int minY = src.height;
  int maxY = 0;
  bool found = false;
  
  for (final pixel in src) {
    if (pixel.a > 0) {
      found = true;
      if (pixel.x < minX) minX = pixel.x;
      if (pixel.x > maxX) maxX = pixel.x;
      if (pixel.y < minY) minY = pixel.y;
      if (pixel.y > maxY) maxY = pixel.y;
    }
  }
  
  if (!found) {
    return src;
  }
  
  // Add padding
  minX = (minX - padding).clamp(0, src.width);
  minY = (minY - padding).clamp(0, src.height);
  maxX = (maxX + padding).clamp(0, src.width);
  maxY = (maxY + padding).clamp(0, src.height);
  
  final width = maxX - minX;
  final height = maxY - minY;
  if (width <= 0 || height <= 0) return src;
  
  return img.copyCrop(src, x: minX, y: minY, width: width, height: height);
}

void main() {
  final sourcePath = r"C:\Users\asus\.gemini\antigravity-ide\brain\195497f5-159f-486a-a73a-294c130a7320\media__1781627568197.jpg";
  final file = File(sourcePath);
  if (!file.existsSync()) {
    print("Source image not found at $sourcePath");
    return;
  }
  
  final bytes = file.readAsBytesSync();
  final image = img.decodeImage(bytes);
  if (image == null) {
    print("Failed to decode image");
    return;
  }
  
  final w = image.width;
  final h = image.height;
  print("Loaded image size: ${w}x${h}");
  
  // Crop top half (Light Mode Logo)
  final topHalf = img.copyCrop(image, x: 0, y: 0, width: w, height: h ~/ 2);
  // Crop bottom half (Dark Mode Logo)
  final bottomHalf = img.copyCrop(image, x: 0, y: h ~/ 2, width: w, height: h - (h ~/ 2));
  
  // Get background pixel from each half
  final lightBg = topHalf.getPixel(5, 5);
  print("Light bg BGR values: R=${lightBg.r}, G=${lightBg.g}, B=${lightBg.b}");
  final logoLight = makeTransparent(topHalf, lightBg, tolerance: 30);
  
  final darkBg = bottomHalf.getPixel(5, 5);
  print("Dark bg BGR values: R=${darkBg.r}, G=${darkBg.g}, B=${darkBg.b}");
  final logoDark = makeTransparent(bottomHalf, darkBg, tolerance: 25);
  
  // Crop tightly
  final logoLightCropped = cropContent(logoLight);
  final logoDarkCropped = cropContent(logoDark);
  
  // Save files
  final lightOut = r"c:\Users\asus\OneDrive\Desktop\eazzio-reminder\frontend\assets\images\logo_light.png";
  final darkOut = r"c:\Users\asus\OneDrive\Desktop\eazzio-reminder\frontend\assets\images\logo_dark.png";
  
  File(lightOut).writeAsBytesSync(img.encodePng(logoLightCropped));
  File(darkOut).writeAsBytesSync(img.encodePng(logoDarkCropped));
  
  print("Saved light logo to: $lightOut");
  print("Saved dark logo to: $darkOut");
}
